### PR TITLE
fix(detectOverflow): remove boundary margin calc

### PR DESCRIPTION
**Pull request by atomiks**
Didn't do much but I think you wanted a PR, merging into the feature branch you have

I think this is the behavior you want as it matches v1 as far as I can tell regarding body/boundary margins

### CODE REVIEW

File: src/modifiers/detectOverflow.js [hunk 1]
```diff
@@ -55,23 +45,22 @@ export function detectOverflow(
     },
   });
 
-  console.log(boundaryMargins);
-
-  const popperClientRect = Object.assign({}, popperOffsets, {
+  const popperClientRect = {
+    ...popperOffsets,
```
Comment (FezVrasta) [inline src/modifiers/detectOverflow.js]: I think I’m trying to avoid to use es next features to reduce the bundle  size

Comment (atomiks) [reply src/modifiers/detectOverflow.js]: So you wanted people to polyfill things like `Object.assign` if they need to? I still don't know how bundlers will be handling modern (~ES2017) and legacy (ES5) code in the future. 

IIRC they want to make ES2017 default on npm and make people transpile `node_modules` in the future to enable differential bundling, or there will be a `targets` field in package.json, not sure 🤔

Comment (atomiks) [reply src/modifiers/detectOverflow.js]: Regardless, I think it would be ES2017 but the spread op is 2018

Comment (FezVrasta) [reply src/modifiers/detectOverflow.js]: Object.assign is well supported, the spread operator when transpiled generates a lot of code

Comment (FezVrasta) [reply src/modifiers/detectOverflow.js]: I just realized I'm using spread operators all around the code, nevermind, ahah

**Pull request merged**